src/validation/rules/NoIntrospectionFieldsRule.js

```diff
@@ -0,0 +1,32 @@
+// @flow strict
+
+import { GraphQLError } from '../../error/GraphQLError';
+
+import { type FieldNode } from '../../language/ast';
+import { type ASTVisitor } from '../../language/visitor';
+
+import { type ValidationContext } from '../ValidationContext';
+
+/**
+ * No introspection fields
+ *
+ * A GraphQL document is only valid if all fields selected are not
+ * fields that return an introspection type. Note: This rule is not
+ * part of the Validation section of the GraphQL Specification.
+ */
+export function NoIntrospectionFieldsRule(
+  context: ValidationContext,
+): ASTVisitor {
+  return {
+    Field(node: FieldNode) {
+      if (node.name.value === '__schema' || node.name.value === '__type') {
```

Comment:
Please check that you are in root type, `context` has functions for that.

Reply:
@IvanGoncharov Fixed!